src/machines/workflow_machines.rs

```diff
@@ -990,6 +973,48 @@ impl WorkflowMachines {
                 MachineResponse::PushWFJob(j) => {
                     jobs.push(j);
                 }
+                MachineResponse::RequestCancelLocalActivity(seq) => {
+                    // We might already know about the status from a pre-resolution. Apply it if so.
+                    // We need to do this because otherwise we might need to perform additional
+                    // activations during replay that didn't happen during execution, just like
+                    // we sometimes pre-resolve activities when first requested.
+                    if let Some(preres) = self.local_activity_data.take_preresolution(seq) {
+                        if let Machines::LocalActivityMachine(lam) = self.machine_mut(m_key) {
+                            let more_responses = lam.try_resolve_with_dat(preres)?;
+                            self.process_machine_responses(m_key, more_responses)?;
+                        } else {
+                            panic!("A non local-activity machine returned a request cancel LA response");
+                        }
+                    }
+                    // If it's in the request queue, just rip it out.
```

Comment:
Is this what happens when a local activity is started and then cancelled before any commands are sent to the server? What does the history look like when the cancel is slow enough to not be in this queue vs when it is? (just curious how it appears)

Reply:
Yes. When the cancel happens before lang knows about the activity, no history entry is created.